File: src/main/java/com/prmncr/normativecontrol/advices/FileSizeExceptionAdvice.java
package com.prmncr.normativecontrol.advices;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.multipart.MaxUploadSizeExceededException;

import java.util.HashMap;
import java.util.Map;

@ControllerAdvice
class FileSizeExceptionAdvice {
    @ExceptionHandler(MaxUploadSizeExceededException.class)
    @ResponseBody
    @ResponseStatus(HttpStatus.BAD_REQUEST)
    public Map<String, Object> handleFileSizeException(MaxUploadSizeExceededException e) {
        var map = new HashMap<String, Object>();
        map.put("status", 400);
        map.put("error", "Document was too big");
        return map;
    }
}

File: src/main/kotlin/com/maeasoftworks/normativecontrol/advices/FileSizeExceptionAdvice.kt
package com.maeasoftworks.normativecontrol.advices

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.multipart.MaxUploadSizeExceededException

@ControllerAdvice
internal class FileSizeExceptionAdvice {
    @ExceptionHandler(MaxUploadSizeExceededException::class)
    @ResponseBody
    @ResponseStatus(HttpStatus.BAD_REQUEST)
    fun handleFileSizeException(e: MaxUploadSizeExceededException?): Map<String, Any> {
        val map = HashMap<String, Any>()
        map["status"] = 400
        map["templates/error"] = "Document was too big"
        return map
    }
}